plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-filter/edit.tsx

```diff
@@ -32,10 +36,13 @@ const Edit = ( props: BlockEditProps< BlockAttributes > ) => {
 		isEditor: true,
 	} );
 
-	useProductFilterClearButtonManager( {
-		clientId,
-		showClearButton: clearButton,
-	} );
+	useEffect( () => {
+		toggleProductFilterClearButtonVisibility( {
+			clientId,
+			showClearButton: clearButton,
+		} );
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [] );
```

Comment:
So we still have to use a hook? It sounds like we're going in a full circle 😂

Reply:
Unfortunately yes 😅, it’s necessary due to the way React handles component lifecycles. The advantage of this new approach is that it will only run when the component mounts for the first time and when the attribute changes. The previous approach was triggered on every render but included a conditional to prevent the code from running every time.